spec/fastsync/vdd/fastsyncL2.tla

```diff
@@ -0,0 +1,676 @@
+  outMsg,                                   \* a node sends a message by this variable
+  nonDetFlag                                \* variable that non-deterministically takes values from the set BOOLEAN    
+  
+  
+(* the variables of the node *)  
+nvars == <<state, blockPool>>
+  
+\* Control messages
+ControlMsgs ==
+    [type: {"addPeer"}, peerId: AllPeerIds]
+        \union
+    [type: {"removePeer"}, peerId: AllPeerIds] 
+        \union
+    [type: {"syncTimeout"}]        
+
+\* All messages (and events) received by a node
+InMsgs ==
+    {NoMsg}
+        \union
+    [type: {"blockResponse"}, peerId: AllPeerIds, block: BlocksWithNil]
+        \union
+    [type: {"statusResponse"}, peerId: AllPeerIds, height: Heights]
+        \union
+    ControlMsgs    
+
+\* Messages sent by a node and received by peers (environment in our case)          
+OutMsgs ==
+    {NoMsg}
+        \union
+    [type: {"statusRequest"}]           \* StatusRequest is broadcast to the set of connected peers.     
+        \union  
+    [type: {"blockRequest"}, peerId: AllPeerIds, height: Heights]  
+        
+
+(********************************** NODE ***********************************)
+
+InitNode == 
+     \E pIds \in SUBSET AllPeerIds \ {{}}:                   \* set of peers node established initial connections with
+        /\ blockPool = [
+                height |-> 1,
+                peerIds |-> pIds, 
+                peerHeights |-> [p \in AllPeerIds |-> NilHeight],     \* no peer height is known
+                blockStore |-> [h \in Heights |-> NilBlock],        
+                receivedBlocks |-> [h \in Heights |-> NilPeer],         
+                pendingBlocks |-> [h \in Heights |-> NilPeer],
+                syncedBlocks |-> -1,
+                syncHeight |-> 1
+           ]
+       /\ state = "running" 
+           
+\* Remove a faulty peer.
+\* Returns new block pool. If the peer is alredy removed, returns input value.  
+RemovePeer(peer, bPool) ==       
+    LET pIds == bPool.peerIds \ {peer} IN
+    LET pHeights == [bPool.peerHeights EXCEPT ![peer] = NilHeight] IN
+    
+    LET failedRequests == 
+        {h \in Heights: /\ h >= bPool.height 
+                        /\ bPool.pendingBlocks[h] = peer
+                        /\ bPool.receivedBlocks[h] = peer} IN
+    LET pBlocks == 
+        [h \in Heights |-> IF h \in failedRequests THEN NilPeer ELSE bPool.pendingBlocks[h]] IN  
+    LET rBlocks == 
+        [h \in Heights |-> IF h \in failedRequests THEN NilPeer ELSE bPool.receivedBlocks[h]] IN
+    LET bStore == 
+        [h \in Heights |-> IF h \in failedRequests THEN NilBlock ELSE bPool.blockStore[h]] IN  
+    
+    IF peer \in bPool.peerIds     
+    THEN [bPool EXCEPT 
+            !.peerIds = pIds,
+            !.peerHeights = pHeights,
+            !.pendingBlocks = pBlocks,
+            !.receivedBlocks = rBlocks,
+            !.blockStore = bStore
+          ]                              
+    ELSE bPool
```

Comment:
Remove the whole operator and replace all calls to `RemovePeer(p, bPool)` with `RemoveManyPeers({p}, bPool)`?